itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestExceptionMessageWithJdbc.java

```diff
@@ -0,0 +1,157 @@
+  private static final String USERNAME = System.getProperty("user.name");
+  private static final String PASSWORD = "bar";
+
+  @BeforeClass
+  public static void setupBeforeClass() throws Exception {
+    MiniHS2.cleanupLocalDir();
+    HiveConf conf = getNewHiveConf();
+    try {
+      startMiniHS2(conf);
+    } catch (Exception e) {
+      System.out.println("Unable to start MiniHS2: " + e);
+      throw e;
+    }
+
+    try {
+      openDefaultConnections();
+    } catch (Exception e) {
+      System.out.println("Unable to open default connections to MiniHS2: " + e);
+      throw e;
+    }
+    Statement stmt = conDefault.createStatement();
+    stmt.execute("drop database if exists " + TEST_DB_NAME + " cascade");
+    stmt.execute("create database " + TEST_DB_NAME);
+    stmt.close();
+
+    try {
+      openTestConnections();
+    } catch (Exception e) {
+      System.out.println("Unable to open default connections to MiniHS2: " + e);
+      throw e;
+    }
+  }
+
+  private static Connection getConnection() throws Exception {
+    return getConnection(miniHS2.getJdbcURL(), USERNAME, PASSWORD);
+  }
+
+  private static Connection getConnection(String dbName) throws Exception {
+    return getConnection(miniHS2.getJdbcURL(dbName), USERNAME, PASSWORD);
+  }
+
+  private static Connection getConnection(String jdbcURL, String user, String pwd)
+      throws SQLException {
+    Connection conn = DriverManager.getConnection(jdbcURL, user, pwd);
+    assertNotNull(conn);
+    return conn;
+  }
+
+  @AfterClass
+  public static void tearDownAfterClass() throws Exception {
+    Statement stmt = conDefault.createStatement();
+    stmt.execute("set hive.support.concurrency = false");
+    stmt.execute("drop database if exists " + TEST_DB_NAME + " cascade");
+    stmt.close();
+    if (conTestDb != null) {
+      conTestDb.close();
+    }
+    if (conDefault != null) {
+      conDefault.close();
+    }
+    stopMiniHS2();
+    cleanupMiniHS2();
+  }
+
+  private static void startMiniHS2(HiveConf conf) throws Exception {
+    MiniHS2.Builder builder = new MiniHS2.Builder().withConf(conf).cleanupLocalDirOnStartup(false);
+    miniHS2 = builder.build();
+    
+    Map<String, String> confOverlay = new HashMap<>();
+    miniHS2.start(confOverlay);
+  }
+
+  private static void stopMiniHS2() {
+    if ((miniHS2 != null) && (miniHS2.isStarted())) {
+      miniHS2.stop();
+    }
+  }
+
+  private static void cleanupMiniHS2() throws IOException {
+    if (miniHS2 != null) {
+      miniHS2.cleanup();
+    }
+    MiniHS2.cleanupLocalDir();
+  }
+
+  private static void openDefaultConnections() throws Exception {
+    conDefault = getConnection();
+  }
+
+  private static void openTestConnections() throws Exception {
+    conTestDb = getConnection(TEST_DB_NAME);
+  }
+
+  private static HiveConf getNewHiveConf() {
+    HiveConf conf = new HiveConfForTest(TestExceptionMessageWithJdbc.class);
+    conf.setVar(ConfVars.HIVE_EXECUTION_ENGINE, "mr");
```

Comment:
Why are we running the tests with MR, MR is deprecated we should avoid adding new tests with it